### PR TITLE
[MLIR][GPUToLLVMSPV] Update ConvertGpuOpsToLLVMSPVOps's option

### PR DESCRIPTION
**Pull request by jle-quel**
## Description

This PR updates the `ConvertGpuOpsToLLVMSPVOps`'s option by replacing the `index-bitwidth` with a boolean option `use-64bit-index` (similar to the `ConvertGPUToSPIRV` option).

The reason for this modification is because the `ConvertGpuOpsToLLVMSPVOps`:
> Generate LLVM operations to be ingested by a SPIR-V backend for gpu operations

In the context of SPIR-V specifications only two physical addressing models are allowed: `Physical32` and `Physical64`.

This change guarantees output sanity by preventing invalid or unsupported index bitwidths from being specified.

### CODE REVIEW

**Review by victor-eds — APPROVED** [review]

**Comment by jle-quel** [comment]
Gentle ping to get your attention @Dinistro @kuhar

**Review by Dinistro — COMMENTED** [review]
I missed this review, sorry for not reacting.

I've dropped a comment, but I'm not really deep into the GPU passes to give you a well founded feedback.

File: mlir/include/mlir/Conversion/Passes.td [hunk 1]
```diff
@@ -553,9 +553,9 @@ def ConvertGpuOpsToLLVMSPVOps : Pass<"convert-gpu-to-llvm-spv", "gpu::GPUModuleO
     "Generate LLVM operations to be ingested by a SPIR-V backend for gpu operations";
   let dependentDialects = ["LLVM::LLVMDialect"];
   let options = [
-    Option<"indexBitwidth", "index-bitwidth", "unsigned",
-           /*default=kDeriveIndexBitwidthFromDataLayout*/"0",
-           "Bitwidth of the index type, 0 to use size of machine word">,
+    Option<"use64bitIndex", "use-64bit-index",
+           "bool", /*default=*/"false",
+           "Use 64-bit integers to convert index types">,
```
[the remark this answers is not in the document]
Comment (jle-quel) [reply mlir/include/mlir/Conversion/Passes.td:558]: Thanks for the feedback!

I understand your reasoning for preferring an **enum** to make the supported options more explicit. However, the related passes—such as `convert-gpu-to-spirv`, `convert-index-to-spirv`, and `convert-memref-to-spirv`—already use this boolean option to toggle between **32-bit** and **64-bit** indices.

To maintain consistency across these conversion passes, would it be better to keep this option as a **boolean** as well? This ensures uniformity and avoids introducing a different style of option specification in a single pass.

**Review by Dinistro — APPROVED** [review]
Considering your explanation, I think that this can be merged as is.

**Pull request merged**